File: app/channels/conversations_channel.rb
class ConversationsChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "conversation_channel"
    stream_from current_user
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  # def send_message(data)
  #   conversation = Conversation.find_by(id: data['conversation_id'])
  #   if conversation && conversation.participates?(current_user)
  #     message = current_user.messages.build({body: data['message']})
  #     message.conversation = conversation
  #     message.save!
  #   end
  # end
end

File: app/javascript/channels/conversations_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("ConversationsChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log("Connected Successfuly")
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    console.log(data)
    document.getElementById("conversation-body").insertAdjacentHTML('beforeend', `
                                  <p>${ data.message.body }</p>
                                  <p>at <strong>${ data.message.created_at }</strong><br>
                                    by <strong>${ data.user }</strong></p>
                                  <hr>
    `)
  }
});
